gnome-initial-setup/gis-driver.c

```diff
@@ -1025,19 +1063,25 @@ gis_driver_shutdown (GApplication *app)
 static void
 gis_driver_init (GisDriver *driver)
 {
+  g_autoptr(GError) error = NULL;
   GisDriverPrivate *priv = gis_driver_get_instance_private (driver);
   GdkScreen *screen;
 
   screen = gdk_screen_get_default ();
 
   priv->small_screen = screen_is_small (screen);
+  priv->startup_sound_uuid = NULL;
 
   load_vendor_conf_file (driver);
 
   g_signal_connect (screen, "size-changed",
                     G_CALLBACK (screen_size_changed), driver);
 
   priv->cancellable = g_cancellable_new ();
+
+  priv->sound_client = eos_hack_sound_client_new (&error);
```

Comment:
We'll need `NULL` checks in the places where this is used.

Reply:
what about?
```
g_return_if_fail (priv->sound_client != NULL);
```

Reply:
```
void
gis_driver_stop_startup_sound (GisDriver *driver)
{
  GisDriverPrivate *priv = gis_driver_get_instance_private (driver);;
  g_return_if_fail (gis_driver_is_hack (driver));
  g_return_if_fail (priv->sound_client != NULL);
  g_return_if_fail (priv->startup_sound_uuid != NULL);

  eos_hack_sound_client_stop (priv->sound_client, priv->startup_sound_uuid,
                              NULL, NULL);
  g_free (priv->startup_sound_uuid);
  g_clear_pointer (&priv->startup_sound_uuid, g_free);
}
```